File: KITE-Simulcast-Test/src/main/java/io/cosmosoftware/kite/simulcast/checks/GaugesCheck.java
package io.cosmosoftware.kite.simulcast.checks;

import io.cosmosoftware.kite.exception.KiteTestException;
import io.cosmosoftware.kite.report.Reporter;
import io.cosmosoftware.kite.simulcast.LoopbackStats;
import io.cosmosoftware.kite.simulcast.pages.SimulcastPageBase;
import io.cosmosoftware.kite.steps.TestStep;
import org.openqa.selenium.WebDriver;

import static io.cosmosoftware.kite.util.ReportUtils.saveScreenshotPNG;
import static io.cosmosoftware.kite.util.ReportUtils.timestamp;

public class GaugesCheck extends TestStep {

  private final SimulcastPageBase loopbackPage;


  private final String rid;
  private final int tid;

  public GaugesCheck(WebDriver webDriver, SimulcastPageBase page, String rid, int tid) {
    super(webDriver);
    this.loopbackPage = page;
    this.rid = rid;
    this.tid = tid;
  }
  
  @Override
  public String stepDescription() {
    return "Gauges values for profile " + rid + tid;
  }
  
  @Override
  protected void step() throws KiteTestException {
    LoopbackStats loopbackStats = loopbackPage.getLoopbackStats();
    Reporter.getInstance().jsonAttachment(report, "stats", loopbackStats.getJson());
    Reporter.getInstance().screenshotAttachment(report,
      "Gauges_" + rid + tid + "_" + timestamp(), saveScreenshotPNG(webDriver));
    loopbackStats.validate(rid, logger);
  }
}

File: KITE-Simulcast-Test/js/checks/GaugesCheck.js
const {TestUtils, TestStep} = require('kite-common');
const {medoozePage} = require('../pages');
/**
 * Class: GaugesCheck
 * Extends: TestStep
 * Description:
 */
class GaugesCheck extends TestStep {
  constructor(kiteBaseTest, rid, tid) {
    super();
    this.driver = kiteBaseTest.driver;
    this.rid = rid;
    this.tid = tid;

    // Test reporter if you want to add attachment(s)
    this.testReporter = kiteBaseTest.reporter;
  }

  stepDescription() {
    return 'Gauges values for profile ' + this.rid + this.tid;
  }

  async step() {

    let loopBackStats = await medoozePage.getLoopbackStats(this);

    // Data
    this.testReporter.textAttachment(this.report, 'stats', JSON.stringify(loopBackStats), "json");

    let screenshot = await TestUtils.takeScreenshot(this.driver);
    this.testReporter.screenshotAttachment(this.report, "Screenshot step", screenshot); 
  }
}

module.exports = GaugesCheck;